### ISSUE TITLE
Update from ESP-IDF v3 to ESP-IDF v4 (stable)

### ISSUE THREAD

**Issue by StevenMHernandez** [StevenMHernandez/ESP32-CSI-Tool#11]
Upgrade to V4.3 (current stable version)

See errors such as #10 

- [x] Upgrade from v3.x to v4.0
- [x] Upgrade from v4.1 to v4.2
- [x] Upgrade from v4.2 to v4.3 (current stable version)
    - [x] Replace `ESP_IF_WIFI_AP` with `WIFI_IF_AP` (See #39)
    - [ ] Update `Component config > Common ESP32-related > UART console baud rate > 1552000` in the README.md (It appears that additional steps are required in v4.3 to set a custom UART console baud rate).
    - [ ] Fix `esp_event_loop.h is deprecated, please include esp_event.h instead`
    - [ ] Fix `void tcpip_adapter_init()' is deprecated`
    - [ ] Fix `esp_err_t esp_event_loop_init(system_event_cb_t, void*)' is deprecated`
    - [ ] Fix `esp_err_t esp_event_loop_init(system_event_cb_t, void*)' is deprecated `

**Comment by Gorgeous2017** [comment]
Did you working for this now? 

I am getting familiar with developing with esp32 recently and I want to use your CSI Tool. My ESP-IDF version is v4.2. I don't want reset to v3 because I don't know what will happen... So many error when I set up the development environment of v4.2 ;(

So I want to know when this will be developed. Thank you!

**Comment by StevenMHernandez** [comment]
Currently, I am not planning to work on this immediately. I have other projects that I need to focus on right now. If anyone is willing to work on this open source project, contributions are always welcome of course!

**Comment by StevenMHernandez** [comment]
OK. Here are some of the changes I made a long while ago to update from v3 to v4 (https://github.com/StevenMHernandez/ESP32-CSI-Tool/tree/espidf_v4). 

It should work, but I have not tested it thoroughly as I mentioned. So if you are interested in testing the functionality to make sure it works and fixing any issues, that would be useful to the project. Thanks

**Comment by pcossutta** [comment]
I just tested the STA. It works but it connects to an IP that is hardcoded in socket_components.h

**Comment by StevenMHernandez** [comment]
Yes. the hardcoded IP is for the AP. Did it cause issues for you? Which version did you test with? Thanks.

I have tested the branch on v4.0 but I have not sat down to make sure it works on higher versions yet.

**Comment by StevenMHernandez** [comment]
Version 4.0 is now available the in the master branch.

**Comment by StevenMHernandez** [comment]
#39 makes it possible to successfully compile the project with ESP-IDF v4.3.

There are a few more tasks that will need to be completed before v4.3 changes are merged into the master branch.

**Comment by StevenMHernandez** [comment]
The project is setup to run with ESP-IDF v4.3.

Thanks @xudesheng for contributing a commit towards this effort 👍
